### PR TITLE
Note that current culture is read-only

### LINKED ISSUE #6938: CultureInfo.CurrentCulture default instance should be documented as readonly
From https://github.com/dotnet/runtime/issues/56405.

In .NET Framework, the default `CultureInfo.CurrentCulture` was mutable. This was changed at some point in .NET Core, but not documented anywhere.

[The existing documentation](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.currentculture) documents a similar change made in .NET 4.6:

image

Ideally the docs should also record this change too.

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #6938.

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/api/system.globalization.cultureinfo.currentculture?view=net-6.0&branch=pr-en-us-7197#remarks).

### CODE REVIEW

File: xml/System.Globalization/CultureInfo.xml [hunk 1]
```diff
@@ -1173,19 +1173,22 @@ If `name` is <xref:System.String.Empty?displayProperty=nameWithType>, the constr
...
 
- The current culture is a property of the executing thread. Retrieving the value of the <xref:System.Globalization.CultureInfo.CurrentCulture%2A?displayProperty=nameWithType> property is a more performant equivalent of retrieving the <xref:System.Globalization.CultureInfo> object returned by the `Thread.CurrentThread.CurrentCulture` property.
+The <xref:System.Globalization.CultureInfo> object that's returned by this property and its associated objects determine the default format for dates, times, numbers, currency values, the sorting order of text, casing conventions, and string comparisons.
+
+The current culture is a property of the executing thread. Retrieving the value of the <xref:System.Globalization.CultureInfo.CurrentCulture%2A?displayProperty=nameWithType> property is a more performant equivalent of retrieving the <xref:System.Globalization.CultureInfo> object returned by the `Thread.CurrentThread.CurrentCulture` property. When you set this property to a <xref:System.Globalization.CultureInfo> object that represents a new culture, the value of the `Thread.CurrentThread.CurrentCulture` property also changes.
```
Comment (tarekgh) [inline xml/System.Globalization/CultureInfo.xml:1184]: I hope we can mention something that `Thread.CurrentThread.CurrentCulture` kind of obsolete and CultureInfo.CurrentCulture should be used instead all the time.

File: xml/System.Globalization/CultureInfo.xml [hunk 2]
```diff
@@ -1173,19 +1173,22 @@ If `name` is <xref:System.String.Empty?displayProperty=nameWithType>, the constr
       </ReturnValue>
       <Docs>
         <summary>Gets or sets the <see cref="T:System.Globalization.CultureInfo" /> object that represents the culture used by the current thread.</summary>
-        <value>An object that represents the culture used by the current thread.</value>
+        <value>The culture used by the current thread.</value>
```
Comment (tarekgh) [inline xml/System.Globalization/CultureInfo.xml:1176]: >The culture used by the current thread [](http://example.com/codeflow?start=15&length=38)

The culture used by the current thread and ask-based async operation.

**Review by tarekgh — APPROVED** [review]
Added a couple of minor comments. LGTM otherwise.

File: xml/System.Globalization/CultureInfo.xml (hunk 2 above)
Comment (gewarren) [inline xml/System.Globalization/CultureInfo.xml:1176]:
```suggestion
        <value>The culture used by the current thread and ask-based asynchronous operations.</value>
```

File: xml/System.Globalization/CultureInfo.xml (hunk 1 above)
Comment (gewarren) [inline xml/System.Globalization/CultureInfo.xml:1184]:
```suggestion
The current culture is a property of the executing thread. When you set this property to a <xref:System.Globalization.CultureInfo> object that represents a new culture, the value of the `Thread.CurrentThread.CurrentCulture` property also changes. However, we recommend that you always use the <xref:System.Globalization.CultureInfo.CurrentCulture?displayProperty=nameWithType> property to retrieve and set the current culture.
```

**Review by gewarren — COMMENTED** [review]
Suggestions from feedback.

File: xml/System.Globalization/CultureInfo.xml [hunk 3]
```diff
@@ -1173,19 +1173,22 @@ If `name` is <xref:System.String.Empty?displayProperty=nameWithType>, the constr
       </ReturnValue>
       <Docs>
         <summary>Gets or sets the <see cref="T:System.Globalization.CultureInfo" /> object that represents the culture used by the current thread.</summary>
```
Comment (gewarren) [inline xml/System.Globalization/CultureInfo.xml:1175]:
```suggestion
        <summary>Gets or sets the <see cref="T:System.Globalization.CultureInfo" /> object that represents the culture used by the current thread and ask-based asynchronous operations.</summary>
```

File: xml/System.Globalization/CultureInfo.xml [hunk 4]
```diff
@@ -1172,16 +1172,16 @@ If `name` is <xref:System.String.Empty?displayProperty=nameWithType>, the constr
         <ReturnType>System.Globalization.CultureInfo</ReturnType>
       </ReturnValue>
       <Docs>
-        <summary>Gets or sets the <see cref="T:System.Globalization.CultureInfo" /> object that represents the culture used by the current thread.</summary>
-        <value>The culture used by the current thread.</value>
+        <summary>Gets or sets the <see cref="T:System.Globalization.CultureInfo" /> object that represents the culture used by the current thread and ask-based asynchronous operations.</summary>
+        <value>The culture used by the current thread and ask-based asynchronous operations.</value>
```
Comment (tarekgh) [inline xml/System.Globalization/CultureInfo.xml:1176]: `Task-based` instead of `ask-based`

**Pull request merged**